lib/rules/no-return-await.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ArrowFunctionExpression(node) {
+                if (node.async && node.body.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+
+                    context.report({
+                        node: sourceCode.getFirstToken(node.body),
+                        loc: node.body.loc,
+                        message: "await in an implicit-return async arrow function is redundant - just return the value directly.",
```

Comment:
I guess just I don't understand how this particular piece of information is relevant. For example, one could theoretically make the error message `` Redundant use of `await` on a return value of 5 in a function called `foo` in another function called `bar` in a file called `baz.js` ``, but that would obviously be unhelpful; most of the information there is irrelevant, and it might mislead the user about what the rule is actually checking for.

The only thing that this rule *is* checking for is the use of `await` on return values; pointing out other features of the code, such as whether the statement is inside an arrow function, seems like it would draw the user's attention away from the main issue.